src/lua/api-gateway/util/OauthClient.lua

```diff
@@ -42,12 +73,20 @@ function OauthClient:makeProfileCall(internalPath, oauth_host)
 
     oauth_host = oauth_host or ngx.var.oauth_host
     ngx.log(ngx.INFO, "profileCall request to host=", oauth_host)
+    local startTime = os.time()
     local res = ngx.location.capture(internalPath, { share_all_vars = true })
+    local endTime = os.time()
+    self:increment(httpCalls)
+
+    local elapsedTime = os.difftime(endTime,startTime) * 1000
+    local elapsedTimeMetric = 'oauth.makeProfileCall.duration'
+    self:time(elapsedTimeMetric, elapsedTime)
 
     local logLevel = ngx.INFO
     if res.status ~= 200 then
         logLevel = ngx.WARN
     end
+    self:increment(httpCalls .. '.makeProfileCall.status.' .. res.status)
```

Comment:
Consider extracting the metric name in a variable.
Move this line of code after self:time function in order to group functionality.